engine/access/rest/websockets/controller.go

```diff
@@ -193,20 +319,57 @@ func (c *Controller) handleListSubscriptions(ctx context.Context, msg models.Lis
 }
 
 func (c *Controller) shutdownConnection() {
-	defer close(c.communicationChannel)
-	defer func(conn *websocket.Conn) {
-		if err := c.conn.Close(); err != nil {
-			c.logger.Error().Err(err).Msg("error closing connection")
+	c.shutdownOnce.Do(func() {
```

Comment:
`sync.Once` has the added functionality that it will block all other callers until the first completes. is that desired here? If not, you can just use an atomic bool with compare and swap

Reply:
[3285b23](https://github.com/onflow/flow-go/pull/6757/commits/3285b23092919ff1d5dde12030f50aab13992904)